dll/win32/browseui/shellfind/CSearchBar.cpp

```diff
@@ -131,6 +136,34 @@ LRESULT CSearchBar::OnSearchButtonClicked(WORD wNotifyCode, WORD wID, HWND hWndC
         return 0;
     }
 
+    // See if we have a szFileName, our first character is not an asterisk,
+    // and we have room in MAX_PATH to add an asterisk at the beginning of szFileName then add it
+    if (wcscmp(pSearchStart->szFileName, L"") != 0)
```

Comment:
Is this change also correct if the filename already contains wildcards?

Reply:
@learn-more, I am not sure about your question. This test is just to determine if the user entered anything in the "All or part of the file name" box. I can do testing against Server 2003 if you give me examples that you think might cause this to fail. Thanks.

Reply:
I think he's asking (not relating to this particular line): even if the user already entered a search expression that contains wildcards, should we assume that the expression is always supposed to match anywhere in the filename?
E.g.
* We know that when the user enters `hello`, we want to search for `*hello*` and find `myhello2.txt`.
* But if the user enters `hello*` (i.e. they clearly understand wildcards), should we only search for files starting with `hello` (and _not_ match `myhello2.txt`), or should we still match `*hello*`?
* What about `hel*lo`?

Reply:
Yes, sorry I forgot to follow this up.

2k3 observed behavior:

![2019-09-28_13-12-01](https://user-images.githubusercontent.com/610685/65815599-f37e0100-e1f1-11e9-9fed-13d3e414f6b8.png)

and

![2019-09-28_13-12-20](https://user-images.githubusercontent.com/610685/65815601-f5e05b00-e1f1-11e9-8a8a-41060c1629cf.png)

Reply:
So this needs to be investigated:
Does it influence extension?
what if the `?` wildcard is used, and so on...

Reply:
OK. Changes made. Thanks.